### ISSUE TITLE
Alternatives to Piper?

### ISSUE THREAD

**Issue by FranklinYu** [jhuangtw/xg2xg#122]
[*Is there an open source equivalent to piper, Google's version control tool?* - Stack Overflow](https://stackoverflow.com/questions/46391415/is-there-an-open-source-equivalent-to-piper-googles-version-control-tool) mentioned that Subversion should be sufficient for small companies, while large companies can use Perforce.

**Comment by aij** [comment]
If you have a repo anywhere near the size of google3, surely you can re-implement piper with a tiny fraction of that code. Then open source it for the rest of us. ;) 

Last I checked, nothing else scales to the same level.

**Comment by FranklinYu** [comment]
That’s true. Meanwhile I thought there might be some Xooglers missing the monorepo.

**Comment by tremayne-stewart** [comment]
True - i just do some funky stuff to get a 'good enough for me' version

**Comment by thiagowfx** [comment]
Not quite, but...

https://git-scm.com/docs/scalar
